### PR TITLE
Fix tree growth/sand issue (typo) and sand<=>sandstone doubling

### PR DESCRIPTION
**Pull request by badtoad2000**
Noticed a couple "sand" issues. The first is merely a typo in your last update, and the second has to do with back-and-forth conversion of sand/sandstone resulting in doubling of said items.

### CODE REVIEW

**Comment by badtoad2000** [comment]
In a previous solution for the sand doubling issue I had proposed keeping the default game's 4:1 Sand:Sandstone ratio, but got a comment that 2:1 was desired to make getting sandstone easier. So I have restored that ratio.

Having slept on this, I wonder why not make sand easier to get.by extending the current crafting of Cobblestone-->Gravel with Gravel-->Sand. This would allow keeping the 4:1 Sand:Sandstone ratio from the default game, while making the crafting of Sandstone easier. Something to consider.

**Comment by Amaz1** [comment]
> Having slept on this, I wonder why not make sand easier to get.by extending the current crafting of Cobblestone-->Gravel with Gravel-->Sand. This would allow keeping the 4:1 Sand:Sandstone ratio from the default game, while making the crafting of Sandstone easier. Something to consider.

:+1:

**Comment by Amaz1** [comment]
Also, if you do implement that idea, if you feel like it, could you also change the can_grow function to the one you posted [here](https://github.com/minetest-LOTT/Lord-of-the-Test/pull/48#issue-130314196) please? 
If you don't want to, that's fine, I can add it (if it's alright with you) at some point.

**Comment by badtoad2000** [comment]
Sure. To be clear you want:
a. 4:1 Sand:Sandstone crafting (or we could keep it 2:1)
b. Craft Gravel to Sand (1:1)
c. and the for loop using ipairs, which should be more maintainable. Not too sure about efficiency as it adds a call to ipairs() and exiting on 1st or 2nd item saves get_item_group calls, but exiting on 3rd item or falling through just adds the cost iterations.

Although a series of IFs would be more real-time efficient, that is:
`
local function can_grow(pos)
        local node = minetest.get_node_or_nil({x = pos.x, y = pos.y, z = pos.z})
        if not node then
                return false
        end
        if minetest.get_item_group(node.name, "soil") ~= 0 then
                return true
        elseif minetest.get_item_group(node.name, "stone") ~= 0 then
                return true
        elseif minetest.get_item_group(node.name, "sand") ~= 0 then
                return true
        end
        return false
end
`

**Comment by badtoad2000** [comment]
Would you be terribly upset if I went and fixed indentation in functions.lua?

**Comment by badtoad2000** [comment]
OK, can_grow only seems to be used by abm for saplings so I guess real-time in not important. Had wondered if map gen might use can_grow, but it appears not.

Is crafting of Mese_crystal > Mese_crystal_shard supposed to be one-way, or should I add a reverse crafting recipe?

I also see stuff like the following:
function lottplants_mallorntree(pos)
    local height = 25 + math.random(5)
                if height < 10 then   **--- this can never be true as 26 <= height >= 30
**

There is an else, but I am wondering what is the intent.

**Comment by Amaz1** [comment]
a) Yeah, change it back to the default one if sand can come from gravel and gravel from cobble.
b) Yeah, and cobble to gravel (1:1)
c) It shouldn't really affect performance. It's just used to make sure that saplings don't destroy chests and other player stuff, which is why it's not called in mapgen.

Mese shards... I'm not sure. You may as well...!

I don't really know what the ```if height < 10 then``` code is for... @fishyWET wrote it, so maybe he could explain what its for?

**Comment by fishyWET** [comment]
tbh, I have no idea either. Wrote it about 2 years ago, likely a mistake i never noticed.

**Comment by badtoad2000** [comment]
Do you all have any issues we me making it so that can_grow allows "air" and "water" as growable? 

Also going to fix an issue with large_roots when the relative position = {0,0,0}. Since the sapling is at {0,0,0}, can_grow always failed. I am also adding a "return true" to large_roots as I am not sure how a bare return is treated.

Will also change the mirksapling abm-function to try smallmirktree when large_roots fails.

**Comment by Amaz1** [comment]
Nope, that sounds great.

Brilliant, thanks!

File: mods/lottplants/functions.lua [hunk 1]
```diff
@@ -586,13 +586,11 @@ function lottplants_smallmirktree(pos)
...
 			end
 			end
-      for i = -1, 1 do
-			for k = -1, 1 do
-          if math.abs(i) + math.abs(k) == 2 then
-            check_add_node({x=pos.x+i,y=pos.y+j,z=pos.z+k},{name="default:jungletree"})
-          end
-      end
-      end
+			for i = -1, 1, 2 do
```
Comment (badtoad2000) [inline mods/lottplants/functions.lua]: This is logically the same, but without needing the math. Part of the fixing indentation I mention...

**Comment by badtoad2000** [comment]
When I merged my new changes to my old branch it updated this pull request. Would you rather have a new request for this?

Also have a few other items I want to change/fix, so was going to do a new pull request for that.
Fix minor issues with:
 - wood shield: wrong armor group
 - copper spear: no reach parameter
 - fences: for fences other than default missing oddly_breakable_by_hand
 - mithril axe: choppy maxlevel should probably be 3

Change where the mobs functions call math.atan(z/x) to math.atan2(z,x). Math.atan2 handles 0 values for x and z, so we can remove 0 checks. It also lets us eliminate some vector tweaks.

Increase looking for water range to 3, to decrease the number of mob suicides (drownings).

**Comment by Amaz1** [comment]
You may as well just use this branch. I can "squash" the commits when I merge it.
Okay, great! Fences are supposed to not be breakable by hand, as is the case with their respective woods. The table bellow should show the minimum tool (choppy level) required to cut the respective woods (I really should make something in-game which has this info...):

|  Hand (no choppy)   | Wood axe (choppy=3)  | Stone axe (choppy=2)  | Steel axe (choppy=1) |
|---|---|---|---|
|  Normal wood & junglewood  | Birch & pine  | Alder  | Lebethron & mallorn  |

Its seems though as if all fences are breakable with a stone axe... So if you could put each in the right choppy group, it would be much appreciated. Everything other than fences and lamps seems to follow this convention...!

**Comment by badtoad2000** [comment]
Planning to remove the "wood" group in the crafted items. So no using doors, tables, chairs in place of wood planks (I assume that was  unintentional). Any reason log stairs/slabs retain the the "tree" group value, or may I remove it as well?

I also notice that lockpick and orc sword have the same crafting recipe, but only lockpicks are craftable. Want me to do anything about that? Should the sword drops for orcs change to orc sword?

In beds.lua, the "lottblocks:bed_bottom_..colour" code appears to be duplicated. Will double check that the code is identical and delete the duplicate.

**Comment by Amaz1** [comment]
Yeah, those things having the wood group was unintentional, as were stairs having the tree group. Feel free to remove them!

It would make sense for orcs to drop the orc sword, so you may as well do that.

When you're happy with the pull, let me know and I'll merge it!

**Comment by badtoad2000** [comment]
Just noticed only raiding orcs have steel swords. Uruk hai drop bronze, and default orcs don't drop weapons. Will change textures to use the orc sword.

Should I make stairs and slabs use parent wood attributes without tree/wood (like chairs, doors, etc)? The "wood" versions are snappy, which doesn't seem quite right.

**Comment by badtoad2000** [comment]
@amaz1 I am happy with the pull.

**Comment by Amaz1** [comment]
Okay!
I've merged it.
https://github.com/minetest-LOTT/Lord-of-the-Test/commit/3373ab23ccf74dd18ba113469744b31006a94240
https://github.com/minetest-LOTT/Lord-of-the-Test/commit/5ae18afda0e0f09ec8f1fe2a6edea7f67f208384
https://github.com/minetest-LOTT/Lord-of-the-Test/commit/a1f7b79f6528b56649eea7f185df4c7647445573
https://github.com/minetest-LOTT/Lord-of-the-Test/commit/67071aa2bdde1346549b7a96a573d75f27923b95

If you want to make any new changes, you might need to either delete your fork of LOTT and make it again, or use some git magic to get your branches in sync with LOTT...

Thanks again!

**Pull request closed without merging**